src/App.js

```diff
@@ -2,32 +2,44 @@ import React, { Component } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
+import { Filter } from './components/Filter';
 
 export class App extends Component {
-  state = {};
+  state = {
+    query: '',
+  };
+
+  onChange = (event) => {
+    this.setState({
+      query: event.target.value,
+    });
+  }
+
+  filterMoviesList = () => (
+    this.state.query.length > 0
+      ? [...moviesFromServer].filter(
+        movie => (
+          movie.title.toLocaleLowerCase()
+            .includes(this.state.query.toLocaleLowerCase())
+          || movie.description.toLocaleLowerCase()
+            .includes(this.state.query.toLocaleLowerCase())
+        ),
+      )
+      : [...moviesFromServer]
```

Comment:
```suggestion
      : moviesFromServer
```
можно передавать сам массив, а не его копию